p2p/src/peers.rs

```diff
@@ -256,7 +257,14 @@ impl Peers {
 			match inner(&p) {
 				Ok(true) => count += 1,
 				Ok(false) => (),
-				Err(e) => debug!("Error sending {} to peer: {:?}", obj_name, e),
+				Err(e) => {
+					debug!(
+						"Error sending {:?} to peer {:?}: {:?}",
+						obj_name, &p.info.addr, e
+					);
+					p.stop();
+					self.peers.write().remove(&p.info.addr);
```

Comment:
Do I understand correctly that we replace the logic of marking peers in `check_connection` with this error handling, assuming if an error happened and we closed the connection on peer's thread exit we will get an error here and remove this peer?

Reply:
Updated description for this PR which should explain what the intent is here.